hooks/useElementPositionAndSize.ts

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from 'react';
+
+export interface ElementPositionAndSize {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+}
+
+export const useElementPositionAndSize = <T extends HTMLElement>(
+  elementRef: React.RefObject<T>
+) => {
+  const [position, setPosition] = useState<ElementPositionAndSize>({
+    x: 0,
+    y: 0,
+    width: 0,
+    height: 0,
+  });
+
+  const getPosition = () => {
+    if (elementRef.current) {
+      const x = elementRef.current.offsetLeft || 0;
+      const y = elementRef.current.offsetTop || 0;
+      const width = elementRef.current.offsetWidth || 0;
+      const height = elementRef.current.offsetHeight || 0;
+
+      setPosition({ x, y, width, height });
+    }
+  };
+
+  useEffect(() => getPosition(), [elementRef.current]);
```

Comment:
```suggestion
  useEffect(() => getPosition(), []);
```
Adding ref in deps wouldn't call the useEffect again

Reply:
But, I think we should actually pass this reference, as we should recalculate if the current element changes, no?

Reply:
but even if we pass the ref in deps, it will not trigger the useEffect and re-calculate. The result will be the same. `getPosition` will be invoked only after mounting by useEffect.(PS: even the eslint rule advises the same to remove the ref from deps)

I believe the recalculation is being happened by the `resize` listener below.

Reply:
ESLint advising on unused props or missing props is not a mechanism of verifying if a prop is needed or not.

You can pass whatever you want on the useEffect "dependencies", as the "dependencies" tell when the effect should be re-run or not.

Sure, I have not checked if they actually use the reference there or if recalculating due to reference change would make sense here, but it's a common practice that if the element you're using as a ref changed and you're making a calculation (even of an element not directly related to that ref) to update something else.

Reply:
Note.: I have no idea if a hook is even needed, I haven't checked what these effects and hooks are doing.